### ISSUE TITLE
[Question] Which is better for Cross Lingual classfication Task: LASER or XLM?

### ISSUE THREAD

**Issue by Ayush-iitkgp** [facebookresearch/LASER#106]
Hello,
I have the training data with labels in English. Now, I want to use this data to predict for other languages. I saw XLM and LASER both support the cross-lingual classification. However, they don't have the benchmark on the same dataset, therefore, it's difficult to know which model is better. Does someone help me in determining which(XLM or LASER) is better for cross-lingual classification?

**Comment by Ayush-iitkgp** [comment]
XLM is better than LASER
1. XLM Benchmark: https://github.com/facebookresearch/XLM#ii-cross-lingual-language-model-pretraining-xlm
2. LASER Benchmark: https://github.com/facebookresearch/LASER/tree/master/tasks/xnli#results

**Comment by PiotrCzapla** [comment]
You can consider further improving the results for regular document classification by following this approach http://nlp.fast.ai/classification/2019/09/10/multifit.html . We used LASER as XLM wasn't available when we were testing multifit. I would be super interested to see how it works with XLM

**Comment by MastafaF** [comment]
XLM does not cover all 100 languages to the best of my knowledge. Which model/implementation did you use @Ayush-iitkgp ?

**Comment by MastafaF** [comment]
Indeed, XLM with MLM+TLM only covers 15 languages currently...

**Comment by Ayush-iitkgp** [comment]
@MastafaF I am starting with the XLM model with 15 languages. XLM does support 100 languages, see [here](https://github.com/facebookresearch/XLM#pretrained-cross-lingual-language-models).

**Comment by MastafaF** [comment]
@Ayush-iitkgp From my reading of their paper a few weeks ago, my understanding is that the version MLM+TLM is the one that gets to the best results in terms of multilingual embeddings and that can outperform LASER. Indeed, multi-BERT already implemented MLM for a large number of languages and the quality of the multilingual embeddings is not optimal.

**Comment by hoschwenk** [comment]
Hello,
it depends on the classification task, and maybe the languages you want to transfer to,

**Comment by Bachstelze** [comment]
@Ayush-iitkgp How is your approach doing? 
I will try classification with model freezing and an extra layer on [XLM-R](https://github.com/pytorch/fairseq/blob/master/examples/xlmr/README.md) for low-ressource languages. [FastText](https://github.com/facebookresearch/fastText) could be interesting if time matters.

**Comment by Ayush-iitkgp** [comment]
@Bachstelze my approach included fine-tuning the XLM model on English data and using zero-shot classification to predict on german and Spanish languages. However, the performance of the model on german and Spanish is less than even 20% (accuracy) so I am still figuring out what can be done. The problem in my case is that I only have labeled English data and a very small amount of non-English data for performance measurement. Do you have any recommendations?

**Comment by Bachstelze** [comment]
How is the performance of the model on English?
The following two recommendations could help with the German and Spanish accuracy problem:
- Preserve the multilingual layers and only train an additional layer on the top. For example set the paramter 'trainable' to 'false' in [allennlp](https://allenai.github.io/allennlp-docs/api/allennlp.models.bert_for_classification.html).
- Translate the labeled data to the other languages. For example with the [transformer.wmt19.en-de](https://github.com/pytorch/fairseq/blob/master/examples/wmt19/README.md)

Hopefully you give the power of your knowledge back to the people.

**Comment by loretoparisi** [comment]
> 1. https://github.com/facebookresearch/XLM#ii-cross-lingual-language-model-pretraining-xlm

Now XML-R has more than 100 language, so it makes a lot of sense to replace it:

> XLM-R is the new state-of-the-art XLM model. XLM-R shows the possibility of training one model for many languages while not sacrificing per-language performance. It is trained on 2.5 TB of CommonCrawl data, in 100 languages
https://github.com/facebookresearch/XLM#ii-cross-lingual-language-model-pretraining-xlm

**Comment by MastafaF** [comment]
Hi @loretoparisi, 
My experiments on the task WMT2012 (similarity search) comparing XLM MLM on 100 languages and LASER shows that LASER clearly outperforms XLM in this case. 
Haven't tried with XLM-R since on my side it is quite buggy still. But I will be happy to share more around it. 

Cheers,

**Comment by loretoparisi** [comment]
@MastafaF pretty interesting test! We did not test XLM-R yet, I wonder how XLM model outperforms LASER bi-LSTM networks, because according to the results they have presented it should be the opposite, but we did not replace LASER anyways for several other reasons. 
At this point, a test with XLM-R must be done!

**Comment by MastafaF** [comment]
Hi @loretoparisi , XLM-R gives poor results at the moment. Stay tuned for further experiments, will post the link soon for replication :)

**Comment by MastafaF** [comment]
Hi @loretoparisi , you can check some tests [here](https://github.com/MastafaF/multilingual_similarity_compare) on WMT2012 reproducing experiments from LASER and doing a comparative study between other multilingual architectures. Feel free to raise an issue or send a PR if need be 😃

**Comment by loretoparisi** [comment]
@MastafaF thank you very much! It's a very comprehensive and rigorous analysis 💯
